op-supervisor/supervisor/backend/status/status.go

```diff
@@ -54,10 +54,21 @@ func (su *StatusTracker) SyncStatus() eth.SupervisorSyncStatus {
 	defer su.mu.RUnlock()
 
 	var supervisorStatus eth.SupervisorSyncStatus
+	// to collect the min synced L1, we need to iterate over all nodes
+	// and compare the current L1 block they each reported.
 	for _, nodeStatus := range su.statuses {
-		if supervisorStatus.MinSyncedL1 == (eth.L1BlockRef{}) || supervisorStatus.MinSyncedL1.Number < nodeStatus.CurrentL1.Number {
+		// if the min synced L1 is not set, or the node's current L1 is lower than the min synced L1,
+		if supervisorStatus.MinSyncedL1 == (eth.L1BlockRef{}) || supervisorStatus.MinSyncedL1.Number > nodeStatus.CurrentL1.Number {
 			supervisorStatus.MinSyncedL1 = nodeStatus.CurrentL1
 		}
+		// if the height is equal, we need to compare the hash
+		if supervisorStatus.MinSyncedL1.Number == nodeStatus.CurrentL1.Number &&
+			supervisorStatus.MinSyncedL1.Hash != nodeStatus.CurrentL1.Hash {
+			// if the hashes are not equal, return an empty status
+			return eth.SupervisorSyncStatus{}
```

Comment:
May be better to return an error instead. So we can easily disambiguate this case from "sync status isn't yet available".

Reply:
I hadn't wanted to change the surface area of existing stuff too much, but now that you mention it, the API already tries to return an error so your suggestion is perfect, thanks